Library/Homebrew/software_spec.rb

```diff
@@ -122,9 +122,12 @@ def option(name, description = "")
         puts "Symbols are reserved for future use, please pass a string instead"
         name = name.to_s
       end
+      unless String === name
```

Comment:
@apjanke You might be interested in reading https://github.com/Homebrew/brew/pull/116#issuecomment-212579880 and preceding comments. Let's try to coordinate this effort (possibly in a new issue/PR)!